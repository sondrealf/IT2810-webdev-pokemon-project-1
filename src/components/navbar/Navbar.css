.navbar__red-bar {
  background-color: #db4032;
  height: 100px;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.navbar__pokemon-ball {
  margin-left: 30px;

  &:hover {
    cursor: pointer;
  }
}

.navbar__button-container {
  display: flex;
  align-items: center;
  height: 100%;
}

.navbar__buttons {
  height: 85%;
  margin-top: 15px;
  border-radius: 15px 15px 0 0;
  color: white;
  width: 140px;
  padding: 0;
  background-color: #db4032;
  border: none;
  font-size: 20px;
  font-weight: 400;

  &:hover {
    background-color: #c22d1f;
    cursor: pointer;
  }

  &:active {
    background-color: #a21f16;
  }

  &:disabled {
    background-color: #f2f2f2;
    cursor: default;
    color: #565656;
  }

  &:focus-visible {
    outline: 4px dashed blue;
  }
}
