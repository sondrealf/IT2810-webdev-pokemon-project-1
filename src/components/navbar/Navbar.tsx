import "./Navbar.css";
import PokemonBall from "../../assets/pokemon-ball.svg";
import { useLocation, useNavigate } from "react-router-dom";

/**
 * A navigation bar component that displays a logo and two buttons for navigating to the home page and the favourites page.
 * @returns {JSX.Element} - A React component that renders a navigation bar.
 */
const Navbar = () => {
  const navigate = useNavigate();
  const location = useLocation();

  return (
    <nav className="navbar__red-bar">
      <img
        onClick={() => navigate("/")}
        src={PokemonBall}
        alt="PokÃ©mon ball"
        className="navbar__pokemon-ball"
      />
      <div className="navbar__button-container">
        <button
          disabled={location.pathname === "/favourites"}
          className="navbar__buttons"
          onClick={() => {
            navigate("/favourites");
            localStorage.setItem("previousPage", "favourites");
          }}
        >
          Favourites
        </button>
        <button
          disabled={location.pathname === "/"}
          className="navbar__buttons"
          onClick={() => navigate("/")}
        >
          Home
        </button>
      </div>
    </nav>
  );
};

export default Navbar;
