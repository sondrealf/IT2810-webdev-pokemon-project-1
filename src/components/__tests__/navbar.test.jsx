import { describe, test, expect } from "vitest";
import Navbar from "../navbar/Navbar";
import { render, screen } from "@testing-library/react";

const mockUseNavigate = vi.fn();
const mockUseLocation = vi.fn();

vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    useNavigate: () => mockUseNavigate,
    useLocation: () => mockUseLocation,
  };
});

describe("Navbar", () => {
  test("matches snapshot", () => {
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });

  test("should render", () => {
    render(<Navbar />);
    expect(screen.getByText("Home")).toBeInTheDocument();
    expect(screen.getByRole("navigation")).toBeInTheDocument();
  });

  test("should navigate to home", () => {
    render(<Navbar />);
    screen.getByText("Home").click();
    expect(mockUseNavigate).toHaveBeenCalledWith("/");
    expect(mockUseNavigate).toHaveBeenCalledTimes(1);
    screen.getByRole("img").click();
    expect(mockUseNavigate).toHaveBeenCalledWith("/");
    expect(mockUseNavigate).toHaveBeenCalledTimes(2);
  });

  test("should navigate to favourites", () => {
    mockUseNavigate.mockClear();
    render(<Navbar />);
    screen.getByText("Favourites").click();
    expect(mockUseNavigate).toHaveBeenCalledWith("/favourites");
    expect(mockUseNavigate).toHaveBeenCalledTimes(1);
  });
});
