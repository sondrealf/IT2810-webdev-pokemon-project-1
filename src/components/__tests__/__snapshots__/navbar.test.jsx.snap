// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > matches snapshot 1`] = `
<div>
  <nav
    class="navbar__red-bar"
  >
    <img
      alt="PokÃ©mon ball"
      class="navbar__pokemon-ball"
      src="/project1/src/assets/pokemon-ball.svg"
    />
    <div
      class="navbar__button-container"
    >
      <button
        class="navbar__buttons"
      >
        Favourites
      </button>
      <button
        class="navbar__buttons"
      >
        Home
      </button>
    </div>
  </nav>
</div>
`;
